app/models/manageiq/providers/redhat/infra_manager/provision_workflow/dialog_field_validation.rb

```diff
@@ -0,0 +1,86 @@
+module ManageIQ::Providers::Redhat::InfraManager::ProvisionWorkflow::DialogFieldValidation
+  def validate_disks_configuration(_field, _values, _dlg, _fld, _value)
+    sparsity = get_value(@values[:disk_sparsity])
+    format = get_value(@values[:disk_format])
+    linked_clone = get_value(@values[:linked_clone])
+    vm_id = get_value(@values[:src_vm_id])
+    vm = VmOrTemplate.find(vm_id)
+
+    if linked_clone
+      if !(default?(sparsity) && default?(format)) || storage_changed_for_disks?(vm)
+        return _("When the 'Linked Clone' option is set, disks and storage configuration cannot be changed")
+      end
+    end
+    disks = vm.disks
+    disks.each do |disk|
+      unless validate_disk(disk)
+        return _("The requested disk configuration for disk #{disk.device_name} is #{requested_disk_sparsity(disk)} with #{requested_disk_format(disk)} format.
+                 This configuration cannot be provisioned on the requested storage")
+      end
+    end
+    nil
+  end
+
+  def requested_disk_format(disk)
+    format = get_value(@values[:disk_format])
+    default?(format) ? disk.format : format
+  end
+
+  def requested_disk_sparsity(disk)
+    sparsity = get_value(@values[:disk_sparsity])
+    disk_sparsity = default?(sparsity) ? disk.thin : sparsity
+    case disk_sparsity
+    when true, "sparse"
+      "sparse"
+    else
+      "preallocated"
+    end
+  end
+
+  def storage_type_from_storage(storage)
+    storage.store_type.to_s.downcase == 'iscsi' ? 'block' : 'file'
+  end
+
+  def storage_type
+    return nil unless dest_storage&.store_type
+
+    storage_type_from_storage(dest_storage)
+  end
+
+  DISK_CONFIGURATIONS = {
+    "file,raw,preallocated"  => true,
```

Comment:
I think instead of constructing a comma separated string this would be slightly better with e.g.
```
storage_supported_format = {"file" => ["raw", "cow"], "block" => ["raw", "cow"]}
storage_supported_sparsity = {"file" => ["sparse"], "block" => ["preallocated", "sparse"]}

storage_supported_format[storage_type].include?(format) && storage_supported_sparsity[storage_type].include?(sparsity)
```

That way if this set grows we don't end up with an `N x M x P` set

Reply:
I am not sure how to do that.
The three variables are not independent - for example the configuration "file,raw,preallocated" works, but "file,raw,sparse" does not, so cow works with file and preallocated works with file, but a combination of the two does not.

I do not think we can fit the information into two arrays with they keys being only one variable, and if we use complex keys seems to be even messier than just straight forward 3 dimensions.

But maybe I am just missing something...

Reply:
Oh okay hm, I'm good with this for now then